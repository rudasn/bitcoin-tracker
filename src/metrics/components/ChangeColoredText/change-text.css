.change-colored-text {
  color: var(--color-muted);
}
.change-colored-text-up {
  color: var(--color-success);
}
.change-colored-text-down {
  color: var(--color-danger);
}
