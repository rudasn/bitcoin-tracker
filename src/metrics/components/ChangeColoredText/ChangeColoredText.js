import React from 'react'
import propTypes from 'prop-types'

import './change-text.css'

const ChangeColoredText = ({ children, change=0 }) => {
  let className

  if (change > 0) {
    className='change-colored-text-up'
  } else if (change < 0) {
    className='change-colored-text-down'
  } else {
    className='change-colored-text'
  }

  return <span className={ className }>{ children }</span>
}

ChangeColoredText.propTypes = {
  change: propTypes.number,
}

export default ChangeColoredText
