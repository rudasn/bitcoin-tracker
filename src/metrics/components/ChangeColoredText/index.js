import React from 'react'

import './change-text.css'

export default ({ children, change }) => {
  let className

  if (change > 0) {
    className='change-colored-text-up'
  } else if (change < 0) {
    className='change-colored-text-down'
  } else {
    className='change-colored-text'
  }

  return <span className={ className }>{ children }</span>
}
