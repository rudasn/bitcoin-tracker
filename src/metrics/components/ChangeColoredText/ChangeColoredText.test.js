import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import ChangeColoredText from './ChangeColoredText'

it('renders without crashing', () => {
  const element = <ChangeColoredText />
  const div = document.createElement('div')
  ReactDOM.render(element, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders positive changes correctly', () => {
  const element = <ChangeColoredText change={ 1 }>Up</ChangeColoredText>
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders negative changes correctly', () => {
  const element = <ChangeColoredText change={ -1 }>Down</ChangeColoredText>
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders no changes correctly', () => {
  const element = <ChangeColoredText change={ 0 }>None</ChangeColoredText>
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})
