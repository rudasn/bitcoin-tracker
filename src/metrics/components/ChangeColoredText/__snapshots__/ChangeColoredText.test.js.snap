// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders negative changes correctly 1`] = `
<span
  className="change-colored-text-down"
>
  Down
</span>
`;

exports[`renders no changes correctly 1`] = `
<span
  className="change-colored-text"
>
  None
</span>
`;

exports[`renders positive changes correctly 1`] = `
<span
  className="change-colored-text-up"
>
  Up
</span>
`;
